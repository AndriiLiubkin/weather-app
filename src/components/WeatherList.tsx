import React from "react";
import SingleDay from "./SingleDay";

import { IWeatherResponseDTO } from "../types";

const WeatherList = ({
  weatherData,
  currentCity,
}: {
  weatherData: IWeatherResponseDTO[];
  currentCity: string;
}) => {
  return (
    <div className="weather-list">
      <div className="current-city">
        <h2>Current City</h2>
        <span>{currentCity}</span>
      </div>
      {weatherData.map((weather: IWeatherResponseDTO) => (
        <SingleDay key={weather.dt} data={weather} />
      ))}
    </div>
  );
};

export default WeatherList;
