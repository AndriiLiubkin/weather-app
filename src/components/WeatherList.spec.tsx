import React from "react";
import { shallow } from "enzyme";
import WeatherList from "./WeatherList";
import Enzyme from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

const mockWeatherListData = [
  {
    dt: 1632495600,
    dt_txt: "2021-09-24 15:00:00",
    main: {
      feels_like: 22.17,
      humidity: 50,
      pressure: 1012,
      temp: 22.55,
    },
    weather: [
      {
        description: "clear sky",
        icon: "01d",
        id: 800,
        main: "Rain",
      },
    ],
    pop: 53,
    visibility: 21,
    wind: {
      deg: 279,
      gust: 7.07,
      speed: 3.92,
    },
  },
  {
    dt: 1632495600,
    dt_txt: "2021-09-24 15:00:00",
    main: {
      feels_like: 22.17,
      humidity: 50,
      pressure: 1012,
      temp: 22.55,
    },
    weather: [
      {
        description: "clear sky",
        icon: "01d",
        id: 800,
        main: "Rain",
      },
    ],
    pop: 53,
    visibility: 21,
    wind: {
      deg: 279,
      gust: 7.07,
      speed: 3.92,
    },
  },
  {
    dt: 1632495600,
    dt_txt: "2021-09-24 15:00:00",
    main: {
      feels_like: 22.17,
      humidity: 50,
      pressure: 1012,
      temp: 22.55,
    },
    weather: [
      {
        description: "clear sky",
        icon: "01d",
        id: 800,
        main: "Rain",
      },
    ],
    pop: 53,
    visibility: 21,
    wind: {
      deg: 279,
      gust: 7.07,
      speed: 3.92,
    },
  },
  {
    dt: 1632495600,
    dt_txt: "2021-09-24 15:00:00",
    main: {
      feels_like: 22.17,
      humidity: 50,
      pressure: 1012,
      temp: 22.55,
    },
    weather: [
      {
        description: "clear sky",
        icon: "01d",
        id: 800,
        main: "Rain",
      },
    ],
    pop: 53,
    visibility: 21,
    wind: {
      deg: 279,
      gust: 7.07,
      speed: 3.92,
    },
  },
  {
    dt: 1632495600,
    dt_txt: "2021-09-24 15:00:00",
    main: {
      feels_like: 22.17,
      humidity: 50,
      pressure: 1012,
      temp: 22.55,
    },
    weather: [
      {
        description: "clear sky",
        icon: "01d",
        id: 800,
        main: "Rain",
      },
    ],
    pop: 53,
    visibility: 21,
    wind: {
      deg: 279,
      gust: 7.07,
      speed: 3.92,
    },
  },
];

Enzyme.configure({ adapter: new Adapter() });

describe("<WeatherList />", () => {
  it("test empty array", () => {
    const wrapper = shallow(<WeatherList weatherData={[]} currentCity="" />);
    expect(wrapper).toMatchSnapshot();
  });

  it("test weather list with data", () => {
    const wrapper = shallow(
      <WeatherList weatherData={mockWeatherListData} currentCity="London" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
