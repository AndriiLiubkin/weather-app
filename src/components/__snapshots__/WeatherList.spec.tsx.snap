// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherList /> test empty array 1`] = `ShallowWrapper {}`;

exports[`<WeatherList /> test weather list with data 1`] = `ShallowWrapper {}`;
